Application/EdFi.Ods.Api/Security/Authorization/AuthorizationSegmentSqlProviderBase.cs

```diff
@@ -94,6 +115,12 @@ public AuthorizationSegmentSqlProviderBase(IAuthorizationViewsProvider authoriza
                         continue;
                     }
 
+                    if (derivedAuthorizationViewName.Contains(authViewName))
```

Comment:
I'm not following the logic for this.  We're coming up with an auth name based on subject to claim, then doing a "contains" check against it for the view name coming back from ViewNameHelper, to determine if we should rename the subject/claim endpoint names.  I'm assuming the contains is to try to work around scenarios with the AuthorizationPathModifier?  We're also modifying the subject and claim endpoint name in some scenarios, so not even sure if this logic path would get hit. This is one of the last tickets so I'd expect us to be moving away from hard coding at this point, but we seem to be doubling down on it further. We need to be writing this to correctly handle transforming the incoming subject and claim names in a clear and consistent way into the appropriate view name and endpoint names in order to drive the later logic for doing the auth.

Reply:
@blmeyers  I updated the code now but still this is correct appoarch or not , I am not sure about it .

Reply:
@blmeyers I missed about scenarios with the AuthorizationPathModifier? ,So I moved logic within public  ClaimsMustBeAssociatedWith method in AuthorizationBuilder.cs